tutorials/Atmos/agnesi_hs_lin.jl

```diff
@@ -0,0 +1,306 @@
+#md #     The following variables are assigned in the initial condition
+#md #     - `state.ρ` = Scalar quantity for initial density profile
+#md #     - `state.ρu`= 3-component vector for initial momentum profile
+#md #     - `state.ρe`= Scalar quantity for initial total-energy profile
+#md #       humidity
+function init_agnesi_hs_lin!(bl, state, aux, (x, y, z), t)
+    # Problem float-type
+    FT = eltype(state)
+
+    # Unpack constant parameters
+    R_gas::FT = R_d(bl.param_set)
+    c_p::FT = cp_d(bl.param_set)
+    c_v::FT = cv_d(bl.param_set)
+    p0::FT = MSLP(bl.param_set)
+    _grav::FT = grav(bl.param_set)
+    γ::FT = c_p / c_v
+
+    c::FT = c_v / R_gas
+    c2::FT = R_gas / c_p
+
+    Tiso::FT = 250.0
+    θ0::FT = Tiso
+
+    Brunt::FT = _grav / sqrt(c_p * Tiso)
+    Brunt2::FT = Brunt * Brunt
+    g2::FT = _grav * _grav
+
+    π_exner::FT = exp(-_grav * z / (c_p * Tiso))
+    θ::FT = θ0 * exp(Brunt2 * z / _grav)
+    ρ::FT = p0 / (R_gas * θ) * (π_exner)^c
+
+    # Compute perturbed thermodynamic state:
+    T = θ * π_exner
+    e_int = internal_energy(bl.param_set, T)
+    ts = PhaseDry(bl.param_set, e_int, ρ)
+
+    #initial velocity
+    u = FT(20.0)
+
+    #State (prognostic) variable assignment
+    e_kin = FT(0)                                       # kinetic energy
+    e_pot = gravitational_potential(bl.orientation, aux)# potential energy
+    ρe_tot = ρ * total_energy(e_kin, e_pot, ts)         # total energy
+
+    state.ρ = ρ
+    state.ρu = SVector{3, FT}(ρ * u, 0, 0)
+    state.ρe = ρe_tot
+end
+
+function setmax(f, xmax, ymax, zmax)
+    function setmaxima(xin, yin, zin)
+        return f(xin, yin, zin; xmax = xmax, ymax = ymax, zmax = zmax)
+    end
+    return setmaxima
+end
+
+function warp_agnesi(xin, yin, zin; xmax = 1000.0, ymax = 1000.0, zmax = 1000.0)
+
+    FT = eltype(xin)
+
+    ac = FT(10000)
+    hm = FT(1)
+    xc = FT(0.5) * xmax
+    zdiff = hm / (FT(1) + ((xin - xc) / ac)^2)
+
+    # Linear relaxation towards domain maximum height
+    x, y, z = xin, yin, zin + zdiff * (zmax - zin) / zmax
+    return x, y, z
+end
+
+# ## [Model Configuration](@id config-helper)
+# We define a configuration function to assist in prescribing the physical
+# model. The purpose of this is to populate the
+# [`ClimateMachine.AtmosLESConfiguration`](@ref LESConfig) with arguments
+# appropriate to the problem being considered.
+function config_agnesi_hs_lin(FT, N, resolution, xmax, ymax, zmax)
+
+    u_relaxation = SVector(FT(20), FT(0), FT(0))
+    c_sponge = 1
+    amp = 1.0
+
+    # Rayleigh damping
+    zsponge = FT(12500.0)
+    rayleigh_sponge =
+        RayleighSponge{FT}(zmax, zsponge, c_sponge, u_relaxation, 2)
+
+    ##SR LSRK144
+    ode_solver = ClimateMachine.ExplicitSolverType(
+        solver_method = LSRK144NiegemannDiehlBusch,
+    )
+
+    source = (Gravity(), rayleigh_sponge)
+
+    temp_profile_ref = DecayingTemperatureProfile{FT}(param_set)
+    ref_state = HydrostaticState(temp_profile_ref)
```

Comment:
Please correct me if I'm wrong, but shouldn't this test case use an isothermal background profile?

Reply:
theta should be uniform in z, but T will still decrease. My question is whether DecayingTemperatureProfile is consistent with the initial background though.